source/Engine.cpp

```diff
@@ -1461,10 +1468,23 @@ void Engine::CalculateStep()
 
 	// Keep track of the flagship to see if it jumps or enters a wormhole this turn.
 	const Ship *flagship = player.Flagship();
+	bool flagshipWasUntargetable = (flagship && !flagship->IsTargetable());
 	bool wasHyperspacing = (flagship && flagship->IsEnteringHyperspace());
-	// Move all the ships.
+	// First, move the player's flagship.
+	MoveShip(player.FlagshipPtr());
+	// Then, move all the rest of the ships.
 	for(const shared_ptr<Ship> &it : ships)
+	{
+		if(it == player.FlagshipPtr())
+			continue;
+		bool wasUntargetable = (!it->IsTargetable());
 		MoveShip(it);
+		// If either the flagship or this ship uncloaked this frame and both are uncloaked,
+		// then they have encountered each other.
+		if(((wasUntargetable && it->IsTargetable()) || (flagshipWasUntargetable && flagship->IsTargetable()))
+			&& flagship->GetSystem() == it->GetSystem() && flagship->IsTargetable())
```

Comment:
```suggestion
		if(((wasUntargetable && it->IsTargetable()) || (flagshipWasUntargetable && flagship->IsTargetable()))
			&& flagship->GetSystem() == it->GetSystem() && flagship->IsTargetable())
```
flagshipIsTargetable can be stored in a variable as well as
flagshipsystem and (flagshipWasUntargetable && flagship->IsTargetable())

Reply:
OK, I see that. Making that change now too.